Gems/AtomLyIntegration/CommonFeatures/Code/Source/ImageBasedLights/EditorImageBasedLightComponent.cpp

```diff
@@ -75,5 +83,74 @@ namespace AZ
         {
         }
 
+
+        AZ::u32 EditorImageBasedLightComponent::OnDiffuseImageAssetChanged()
+        {
+            ImageBasedLightComponentConfig& configuration = m_controller.m_configuration;
+
+            if (UpdateImageAsset(configuration.m_diffuseImageAsset, "_ibldiffuse", "Diffuse",
+                                 configuration.m_specularImageAsset, "_iblspecular", "Specular"))
+            {
+                m_controller.SetSpecularImageAsset(configuration.m_specularImageAsset);
+            }
+
+            return AZ::Edit::PropertyRefreshLevels::ValuesOnly;
+        }
+
+        AZ::u32 EditorImageBasedLightComponent::OnSpecularImageAssetChanged()
+        {
+            ImageBasedLightComponentConfig& configuration = m_controller.m_configuration;
+
+            if (UpdateImageAsset(configuration.m_specularImageAsset, "_iblspecular", "Specular",
+                                 configuration.m_diffuseImageAsset, "_ibldiffuse", "Diffuse"))
+            {
+                m_controller.SetDiffuseImageAsset(configuration.m_diffuseImageAsset);
+            }
+
+            return AZ::Edit::PropertyRefreshLevels::ValuesOnly;
+        }
+
+        bool EditorImageBasedLightComponent::UpdateImageAsset(
+            Data::Asset<RPI::StreamingImageAsset>& asset1,
+            const char* asset1Suffix,
+            const char* asset1Name,
+            Data::Asset<RPI::StreamingImageAsset>& asset2,
+            const char* asset2Suffix,
+            const char* asset2Name)
+        {
+            AZStd::string assetPath = asset1.GetHint();
+            AZ::Data::AssetId assetId;
+
+            if (!assetPath.empty())
+            {
+                // try to load the matching image asset by replacing the asset1 suffix with the asset2 suffix
+                AZ::StringFunc::Replace(assetPath, asset1Suffix, asset2Suffix);
```

Comment:
I recommend checking the asset ID first and getting the path from the asset ID instead of using the hint because the hint will not always be set.

Reply:
Done.